### PR TITLE
Add nodejs18 in function controller

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:

- define nodejs18 as allowed runtime
- switch unit tests to use nodejs18

**Related issue(s)**
#16765 
Depends on https://github.com/kyma-project/kyma/pull/16768

### CODE REVIEW

File: components/function-controller/pkg/apis/serverless/v1alpha1/function_types.go [hunk 1]
```diff
@@ -13,18 +13,19 @@ const (
...
 	Nodejs14 Runtime = "nodejs14"
 	Nodejs16 Runtime = "nodejs16"
+	Nodejs18 Runtime = "nodejs18"
 	Python39 Runtime = "python39"
 )
 
 // RuntimeExtended enumerates runtimes that are either currently supported or
 // no longer supported but there still might be "read-only" Functions using them
-// +kubebuilder:validation:Enum=nodejs12;nodejs14;nodejs16;nodejs10;python38;python39
+// +kubebuilder:validation:Enum=nodejs12;nodejs14;nodejs16;nodejs18;nodejs10;python38;python39
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply components/function-controller/pkg/apis/serverless/v1alpha1/function_types.go:28]: done

**Review by dbadura — DISMISSED** [review]

**Review by dbadura — APPROVED** [review]

**Review by dbadura — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Review by pbochynski — APPROVED** [review]

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**